Light Enemy (In Progress)

JoshCTFS opened the pull request:
Light Enemy Script (Not Yet Tested)
Temp Med Enemy Script

Temp LightEnemy Prefab made without animations

Renamed some of the sprites appropriately (Subject to change)

Also have made New Layer labeled "Player" for ease of visibility as well as other functions.

JoshCTFS committed:
Light Enemy (In Progress)

Light Enemy Script (Not Yet Tested)
Temp Med Enemy Script

Temp LightEnemy Prefab made without animations

Renamed some of the sprites appropriately (Subject to change)

MV1090 commented on Assets/_Scripts/Enemy/LightEnemy.cs:
```diff
@@ -0,0 +1,71 @@
+            if (other.CompareTag("PlayerProjectile"))
+            {
+                damageTaken++;
+                if (damageTaken >= currentHealth)
```
This logic is taken care of in the Enemy script, on collision you just need to call TakeDamage() and input the amount of damage taken.

MV1090 commented on Assets/_Scripts/Enemy/LightEnemy.cs:
```diff
@@ -0,0 +1,71 @@
+                damageTaken++;
+                if (damageTaken >= currentHealth)
+                {
+                Destroy(obj: LightEnemy); 
```
The enemy is destroyed from the  EnemyDeath(input score value) method which needs to be called as an animation event at the end of the death animation.

MV1090 commented on Assets/_Scripts/Enemy/MediumEnemy.cs:
```diff
@@ -0,0 +1,71 @@
+    {
+        if (other.CompareTag("PlayerProjectile"))
+        {
+            damageTaken++;
```
Same as above

MV1090 commented on Assets/_Scripts/Enemy/MediumEnemy.cs:
```diff
@@ -0,0 +1,71 @@
+            damageTaken++;
+            if (damageTaken >= currentHealth)
+            {
+                Destroy(obj: MediumEnemy);
```
Same as above

MV1090 reviewed (commented):
Good work bud, you just need to add the Shoot() method, and the enemy killed method to the end of the death animation when its in. I like the strafing I completely forgot about this, as its the same for all the enemy's it might be better off on the Enemy class so we don't need to repeat it on all the scripts..... Ziyad what do you think of this idea? or is the strafing taken care of somewhere else?

SolomanDavis commented:
@MV1090 @JoshCTFS re: strafing
This strafing code is going to be super useful to be used in combination with the enemy pathfinding hover state. (see https://github.com/SolomanDavis/VGP145-S24-capstone-retro/pull/35).

However few things to note:
When the enemy is in the `EnemyPathfinding.Hover` state, the enemy will simply move towards its assigned `HoverLocation`. This means that the enemy itself is not the entity doing the strafing. Rather it is the assigned `HoverLocation` and its container (the upcoming rank grid) that will be doing the strafing and where this code should be applied.

In terms of where the strafing code should specifically live - I believe it should live in an `EnemyRankGrid` class that I'll be implementing today with @Armaan059 since it's the grid itself that will strafe, while the enemies are sticking to the assigned slots within the grid.

JoshCTFS commented:
> @MV1090 @JoshCTFS re: strafing This strafing code is going to be super useful to be used in combination with the enemy pathfinding hover state. (see #35).
> 
> However few things to note: When the enemy is in the `EnemyPathfinding.Hover` state, the enemy will simply move towards its assigned `HoverLocation`. This means that the enemy itself is not the entity doing the strafing. Rather it is the assigned `HoverLocation` and its container (the upcoming rank grid) that will be doing the strafing and where this code should be applied.
> 
> In terms of where the strafing code should specifically live - I believe it should live in an `EnemyRankGrid` class that I'll be implementing today with @Armaan059 since it's the grid itself that will strafe, while the enemies are sticking to the assigned slots within the grid.

Ah, my apologies. oversight on my part! Thanks for the input!

JoshCTFS committed:
Update to LightEnemy & MediumEnemy Scripts

See above

JoshCTFS committed:
Git Ignore Update

Git Ignore Update

JoshCTFS committed:
Fixed Git Ignore AGAIN

Git Ignore

JoshCTFS committed:
Meta Fix

See above

JoshCTFS committed:
Updated animations

See above

JoshCTFS committed:
Merge branch 'master' into JoshC

# Conflicts:
#	Assets/Scenes/Game.unity
#	Assets/Sprites/Characters/image_2024-04-18_152227724-removebg-preview.png.meta
#	ProjectSettings/TagManager.asset

JoshCTFS committed:
Light Enemy, Medium Enemy updated and Animations placed.

See above^

JoshCTFS committed:
Light Enemy Script and Medium Enemy Script Updated

Commented out strafing so as to preserve for potential future use.
Also changed parameters on LightEnemy and MediumEnemy to EnemyHP, and edited the corresponding scripts.

JoshCTFS committed:
Unity File Update

JoshCTFS committed:
Unity file update

JoshCTFS committed:
Placeholder for Enemy Diving and score change

See above

JoshCTFS committed:
Merge branch 'master' into JoshC

# Conflicts:
#	Assets/Scenes/Game.unity
#	Assets/_Prefabs/HeavyEnemy.prefab
#	ProjectSettings/TagManager.asset

JoshCTFS committed:
Light and Medium Enemy Updated

Light and Medium Enemy OnTrigger function updated to coordinate with Enemy class.
Scores also updated as well as pathfinding functionality from Heavy implemented for future use.

Attempted Switch case but alas
